src/init.cpp

```diff
@@ -1078,7 +1086,7 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
 
     // Start the lightweight task scheduler thread
     CScheduler::Function serviceLoop = boost::bind(&CScheduler::serviceQueue, &scheduler);
-    threadGroup.create_thread(boost::bind(&TraceThread<CScheduler::Function>, "scheduler", serviceLoop));
+    scheduler_thread = std::thread(std::bind(&TraceThread<CScheduler::Function>, "scheduler", serviceLoop));
```

Comment:
Nit:

I find the semantics of bind to be pretty confusing confusing. If we're cleaning up threading semantics I think it might be clearer to do as follows:

scheduler_thread = std::thread([](){ TraceThread("scheduler", CScheduler::serviceQueue(scheduler)); };

This should also allow you to get rid of the explicit template param.